rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceModuleImpl.java

```diff
@@ -166,4 +186,63 @@ private Block getByJsonBlockId(String id) {
             }
         }
     }
+
+    private List<TransactionTrace> buildBlockTraces(Block block, TraceFilterRequest traceFilterRequest) {
+        List<TransactionTrace> blockTraces = new ArrayList<>();
+
+        if (block.getNumber() != 0) {
+            List<Transaction> txList = block.getTransactionsList();
+
+            ProgramTraceProcessor programTraceProcessor = new ProgramTraceProcessor();
+
+            if (traceFilterRequest != null) {
+                Stream<Transaction> txStream = block.getTransactionsList().stream();
+
+                if (traceFilterRequest.getFromAddress() != null && !traceFilterRequest.getFromAddress().isEmpty()) {
+                    List<BigInteger> addresses = traceFilterRequest.getFromAddressAsBigIntegers();
+                    txStream = txStream.filter(tx -> addresses.contains(HexUtils.forceParseStringHexToBigInteger(tx.getSender().toHexString())));
+                }
+
+                if (traceFilterRequest.getToAddress() != null && !traceFilterRequest.getToAddress().isEmpty()) {
+                    List<BigInteger> addresses = traceFilterRequest.getToAddressAsBigIntegers();
+                    txStream = txStream.filter(tx -> addresses.contains(HexUtils.forceParseStringHexToBigInteger(tx.getReceiveAddress().toHexString())));
+                }
+
+                txList = txStream.collect(Collectors.toList());
+            }
+
+            for (Transaction tx : txList) {
+                TransactionInfo txInfo = receiptStore.getInMainChain(tx.getHash().getBytes(), this.blockStore).orElse(null);
+                Objects.requireNonNull(txInfo);
+                txInfo.setTransaction(tx);
+
+                SummarizedProgramTrace programTrace = (SummarizedProgramTrace) programTraceProcessor.getProgramTrace(tx.getHash());
+
+                if (programTrace == null) {
+                    blockTraces.clear();
+                    return new ArrayList<>();
```

Comment:
This return represents some kind of "exception" or "corner case"? If so, we should add a log maybe.

Reply:
No, just means there no traces at all for the transactions we were iterating, should return empty